### PR TITLE
Update parse_type.rs test

### PR DESCRIPTION
**Pull request by seniorjoinu**
Just a typo fix.
It is impossible to construct such type.

### CODE REVIEW

File: candid/tests/parse_type.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ fn parse_idl_prog() {
     let prog = r#"
 import "test.did";
 type my_type = nat8;
-type List = record { head: int; tail: List };
+type List = record { head: int; tail: opt List };
```
Comment (chenyan-dfinity) [inline candid/tests/parse_type.rs:15]:
```suggestion
type List = variant { head: int; tail: List };
```

Comment (rossberg) [reply candid/tests/parse_type.rs]: This change doesn't make sense to me. That variant doesn't define a list, whereas the record did define a (non-empty) list.

**Comment by rossberg** [comment]
@chenyan-dfinity:
> The spec actually allows type A = record { a: A }. This is equivalent to the Empty type, which has no value.

Nit: they are isomorphic (in Candid), but not equivalent. ;)

**Comment by seniorjoinu** [comment]
I was thinking of your suggested change (opt -> variant).
While this type can be constructed, it is not a "list" anymore - one only can define "head" or "tail" but not both. 
I think we should revert to `opt`.

File: candid/tests/parse_type.rs [hunk 2]
```diff
@@ -12,7 +12,7 @@ fn parse_idl_prog() {
     let prog = r#"
 import "test.did";
 type my_type = nat8;
-type List = record { head: int; tail: List };
+type List = variant { head: int; tail: List };
```
Comment (nomeata) [inline candid/tests/parse_type.rs:15]: I guess this should be
```suggestion
type List = opt record { head: int; tail: List };
```

Comment (seniorjoinu) [reply candid/tests/parse_type.rs:15]: It is very interesting, how many forms can one simple data structure have.
I vote for this.

File: candid/tests/parse_type.rs (hunk 1 above)
Comment (chenyan-dfinity) [reply candid/tests/parse_type.rs]: Right, I wasn't thinking. As @nomeata suggested, it should be `opt record` instead.

**Pull request merged**